File: Form_DateRange.vb
Public Class Form_DateRange
    Public ReadOnly Property StartDatePicker As DateTimePicker
        Get
            Return DateTimePicker1
        End Get
    End Property

    Public ReadOnly Property EndDatePicker As DateTimePicker
        Get
            Return DateTimePicker2
        End Get
    End Property

    Private Sub ButtonOK_Click(sender As Object, e As EventArgs) Handles ButtonOK.Click
        ' OKボタンがクリックされたら、DialogResultをOKに設定し、フォームを閉じる
        Me.DialogResult = DialogResult.OK
        Me.Close()
    End Sub

    Private Sub ButtonClear_Click(sender As Object, e As EventArgs) Handles ButtonClear.Click
        ' キャンセルボタンがクリックされたら、DialogResultをキャンセルに設定し、フォームを閉じる
        Me.DialogResult = DialogResult.Cancel
        Me.Close()
    End Sub

    Private Sub Form_DateRange_Load(sender As Object, e As EventArgs) Handles Me.Load

    End Sub


End Class

File: src/Form_DateRange.cs
using System;
using System.Windows.Forms;

namespace TmCGPTD
{
    public partial class Form_DateRange
    {
        public DateTimePicker StartDatePicker
        {
            get
            {
                return DateTimePicker1;
            }
        }

        public DateTimePicker EndDatePicker
        {
            get
            {
                return DateTimePicker2;
            }
        }

        public Form_DateRange()
        {
            InitializeComponent();
        }

        private void ButtonOK_Click(object sender, EventArgs e)
        {
            // OKボタンがクリックされたら、DialogResultをOKに設定し、フォームを閉じる
            DialogResult = DialogResult.OK;
            Close();
        }

        private void ButtonClear_Click(object sender, EventArgs e)
        {
            // キャンセルボタンがクリックされたら、DialogResultをキャンセルに設定し、フォームを閉じる
            DialogResult = DialogResult.Cancel;
            Close();
        }

        private void Form_DateRange_Load(object sender, EventArgs e)
        {

        }


    }
}